libraries/Wire/src/Wire.cpp

```diff
@@ -327,17 +324,18 @@ size_t TwoWire::write(const uint8_t *data, size_t quantity)
 
   if (transmitting) {
     // in master transmitter mode
-    allocateTxBuffer(txBufferLength + quantity);
-    // error if no memory block available to allocate the buffer
-    if (txBuffer == nullptr) {
+    if (allocateTxBuffer(txDataSize + quantity) == 0) {
+      ret = 0;
+    } else if (txBuffer == nullptr) {
```

Comment:
I wonder if this test is needed as if allocateTxBuffer failed it reach Error_Handler ?

Reply:
Right, I remove it for both TX and RX